src/cls/tabular/fbwriter_fbu.cc

```diff
@@ -0,0 +1,932 @@
+    // flush to write whenever cols_per_fb divides i.
+    std::vector< flatbuffers::Offset< Tables::Col_FBU > > cols_vect ;
+    for( unsigned int i = 0; i < ncols; i++ ) {
+      auto col_name     = schema[ i ] ;
+      uint8_t col_index = (uint8_t)i ;
+
+      auto key   = filedata.indexer[i][0] ;
+      auto index = filedata.indexer[i][1] ;
+
+      if( debug )
+        std::cout << "processing col " << std::to_string( i ) << std::endl ;
+
+      if( boost::algorithm::ends_with( key, std::to_string( Tables::SDT_UINT64 ) ) ) {
+        std::vector< uint64_t > int_vect = filedata.listof_int_vect_raw[ std::stoi(index) ] ;
+        auto int_vect_fb = builder.CreateVector( int_vect ) ;
+        auto data = Tables::CreateSDT_UINT64_FBU( builder, int_vect_fb ) ;
+        auto col = Tables::CreateCol_FBU(
+          builder,                              //builder
+          rids_vect_fb,                         //rids
+          nullbits_vector_fb,                   //nullbits
+          col_name,                             //col_name
+          col_index,                            //col_index
+          Tables::DataTypes_FBU_SDT_UINT64_FBU, //data_type
+          data.Union() ) ;                      //data
+        cols_vect.push_back( col ) ;
+      }
+      else if( boost::algorithm::ends_with( key, std::to_string( Tables::SDT_FLOAT ) ) ) {
+        std::vector< float > float_vect = filedata.listof_float_vect_raw[ std::stoi(index) ] ;
+        auto float_vect_fb = builder.CreateVector( float_vect ) ;
+        auto data = Tables::CreateSDT_FLOAT_FBU( builder, float_vect_fb ) ;
+        auto col = Tables::CreateCol_FBU(
+          builder,                              //builder
+          rids_vect_fb,                         //rids
+          nullbits_vector_fb,                   //nullbits
+          col_name,                             //col_name
+          col_index,                            //col_index
+          Tables::DataTypes_FBU_SDT_FLOAT_FBU,  //data_type
+          data.Union() ) ;                      //data
+        cols_vect.push_back( col ) ;
+      }
+      else if( boost::algorithm::ends_with( key, std::to_string( Tables::SDT_STRING ) ) ) {
+        std::vector< flatbuffers::Offset<flatbuffers::String> > string_vect = filedata.listof_string_vect_raw[ std::stoi(index) ] ;
+        auto string_vect_fb = builder.CreateVector( string_vect ) ;
+        auto data = Tables::CreateSDT_STRING_FBU( builder, string_vect_fb ) ;
+        auto col = Tables::CreateCol_FBU(
+          builder,                              //builder
+          rids_vect_fb,                         //rids
+          nullbits_vector_fb,                   //nullbits
+          col_name,                             //col_name
+          col_index,                            //col_index
+          Tables::DataTypes_FBU_SDT_STRING_FBU, //data_type
+          data.Union() ) ;                      //data
+        cols_vect.push_back( col ) ;
```

Comment:
This is adding an entire col of a single data type to the cols vector here right?  If so, cool then we should definitely consider adding compression here later.  What do you think should we compress, the "col" or the "string_vec" or the "string_vec_fb"? Or will those fail so instead we need to compress the "data"?